src/smolvm/facade.py

```
@@ -1485,6 +1540,59 @@ def __exit__(self, *args: object) -> None:
+        """Mount 9p workspace shares with overlayfs inside the guest."""
+        assert self._ssh is not None  # noqa: S101 — caller guarantees SSH ready
+
+        workspace_mounts = self._info.config.workspace_mounts
+        if not workspace_mounts:
+            return
+
+        for index, ws in enumerate(workspace_mounts):
+            tag = ws.mount_tag or f"workspace{index}"
+            guest_path = shlex.quote(ws.guest_path)
+            lower = shlex.quote(f"/mnt/.smolvm-ws-{tag}")
+            upper = shlex.quote(f"/tmp/.smolvm-ws-{tag}-upper")
```

Ruff (0.15.9)

[error] Probable insecure usage of temporary file or directory: "/tmp/.smolvm-ws-"

(S108)